lib/ezutils/classes/ezextension.php

```diff
@@ -544,10 +544,10 @@ public static function getHandlerClass( ezpExtensionOptions $options )
             // we rely on the autoload system here
             if ( class_exists( $handler ) )
             {
-                // only use reflection if we have params to avoid exception on objects withouth constructor
-                if ( $handlerParams !== null && is_array( $handlerParams ) && count( $handlerParams ) > 0 )
+                $reflection = new ReflectionClass( $handler );
+                // only pass parameters if the class has a constructor
+                if ( $reflection->getConstructor() !== null && $handlerParams !== null && is_array( $handlerParams ) && count( $handlerParams ) > 0 )
                 {
-                    $reflection = new ReflectionClass( $handler );
```

Comment:
I think we need to make user aware of this so it does not silently allow people to forget constructor, could you try something like:
```php
                {
                    $reflection = new ReflectionClass( $handler );
                    if ( $reflection->getConstructor() !== null )
                    {
                        $object = $reflection->newInstanceArgs( $handlerParams );
                    }
                    else
                    {
                        eZDebug::writeWarning( 'Constructor is missing but parameters are provided to class ' . $handler . " as defined in setting $iniFile [$iniSection] $iniVariable", __METHOD__ );
                        $object = new $handler();
                    }
```